src/main/java/org/jenkinsci/plugins/nomad/NomadApi.java

```diff
@@ -36,25 +38,68 @@
         return jobs;
     }
 
+    /**
+     * Executes a given request and returns the response body. <b>Note:</b> This method reuses the underlying http client but in case of
+     * an error, this client gets destroyed and recreated when this method is called again.
+     * @param request Any request (not null)
+     * @return Response body as String or an empty String. (not null)
+     */
     String checkResponseAndGetBody (Request request) {
         String bodyString = "";
-        try ( Response response = client.newCall(request).execute();
-              ResponseBody body = response.body())
-        {
-            bodyString = body.string();
-
-            if (response.code() != 200)
-            {
-                LOGGER.log(Level.SEVERE, bodyString, bodyString);    
+        try (Response response = client().newCall(request).execute();
+             ResponseBody responseBody = response.body();
+        ) {
+            bodyString = responseBody.string();
+            if (!response.isSuccessful()) {
+                LOGGER.log(Level.SEVERE, "Request was not successful! Code: "+response.code()+", Body: '"+bodyString+"'");
+                if (Arrays.asList(401, 403, 500).contains(response.code())) {
+                    resetClient();
+                }
             }
         } catch (IOException e) {
-            LOGGER.log(Level.SEVERE, e.getMessage() + "\nRequest:\n" + request.toString());
+            LOGGER.log(Level.SEVERE, e.getMessage() + "\nRequest:\n" + request);
+            resetClient();
         } catch (NullPointerException e) {
-            LOGGER.log(Level.SEVERE, "Error: Got no Nomad response." + "\nRequest:\n" + request.toString());    
+            LOGGER.log(Level.SEVERE, "Error: Got no Nomad response." + "\nRequest:\n" + request.toString());
         }
         return bodyString;
     }
 
+    /**
+     * Reset client so that a new client gets created when {@link #client()} is called.
+     */
+    private void resetClient() {
+        client = null;
+        LOGGER.log(Level.INFO, "Client has been reset!");
+    }
+
+    /**
+     * Provides an {@link OkHttpClient} instance. The existing client is reused. <b>Note</b> The client initialization is synchronized
+     * which means this method is thread safe and can be called from different threads in parallel.
```

Comment:
I think this is enough:

```suggestion
     * Provides an {@link OkHttpClient} instance. Create a new client or reuse the existing one.
     * This method is threadsafe.
```